charts/vineyard/templates/worker/statefulset.yaml

```diff
@@ -0,0 +1,165 @@
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+{{- $name := include "vineyard.name" . }}
+{{- $fullName := include "vineyard.fullname" . }}
+{{- $chart := include "vineyard.chart" . }}
+apiVersion: apps/v1
+kind: StatefulSet
+metadata:
+  name: {{ $fullName }}-worker
+  labels:
+    app: {{ $name }}
+    chart: {{ $chart }}
+    release: {{ .Release.Name }}
+    heritage: {{ .Release.Service }}
+    role: vineyard-worker
+    fluid.io/dataset: {{ .Release.Namespace }}-{{ .Release.Name }}
+  {{- if .Values.owner.enabled }}
+  ownerReferences:
+  - apiVersion: {{ .Values.owner.apiVersion }}
+    blockOwnerDeletion: {{ .Values.owner.blockOwnerDeletion }}
+    controller: {{ .Values.owner.controller }}
+    kind: {{ .Values.owner.kind }}
+    name: {{ .Values.owner.name }}
+    uid: {{ .Values.owner.uid }}
+  {{- end }}
+spec:
+  replicas: {{ .Values.worker.replicas }}
+  serviceName: {{ $fullName }}-worker-svc
+  podManagementPolicy: Parallel
+  selector:
+    matchLabels:
+      app: {{ $name }}
+      chart: {{ $chart }}
+      release: {{ .Release.Name }}
+      role: vineyard-worker
+  template:
+    metadata:
+      annotations:
+        sidecar.istio.io/inject: "false"
+      labels:
+        app: {{ $name }}
+        chart: {{ $chart }}
+        release: {{ .Release.Name }}
+        heritage: {{ .Release.Service }}
+        role: vineyard-worker
+        fluid.io/dataset: {{ .Release.Namespace }}-{{ .Release.Name }}
+    spec:
+      {{- if .Values.worker.nodeSelector }}
+      nodeSelector:
+{{ toYaml .Values.worker.nodeSelector | trim | indent 8  }}
+      {{- end }}
+      containers:
+        - name: vineyard-worker
+          image: {{ .Values.worker.image }}:{{ .Values.worker.imageTag }}
+          imagePullPolicy: {{ .Values.worker.imagePullPolicy }}
+          command:
+          - /bin/bash
+          - -c
+          - >
+            /usr/local/bin/vineyardd
+            --socket /var/run/vineyard/vineyard.sock
+            --size {{ include "vineyard.worker.size" . }}
+            {{- if eq (toString (index .Values.worker.options "vineyardd.reserve.memory")) "true" }}
+            --reserve_memory
+            {{- end }}
+            --meta_timeout {{ default "120" (index .Values.worker.options "wait.etcd.timeout") }}
+            --etcd_prefix {{ default "/vineyard" (index .Values.worker.options "etcd.prefix") }}
+            --etcd_endpoint {{ include "vineyard.master.endpoint" . }}
+            {{- if eq (include "vineyard.checkTieredStore" .) "true" }}
+            --spill_path {{ include "vineyard.spill.path" . }}
+            --spill_lower_rate {{ include "vineyard.spill.lowerRate" . }}
+            --spill_upper_rate {{ include "vineyard.spill.upperRate" . }}
+            {{- end }}
```

Comment:
I think better to directly use `/usr/local/bin/vineyard` as the entrypoint command. Seems that `bash -c` is not necessary here.

Reply:
Done.